bananas_api/new_upload/classifiers/scenario.py

```diff
@@ -0,0 +1,69 @@
+from ...helpers.api_schema import Classification
+from ...helpers.enums import (
+    Climate,
+    Resolution,
+    Shape,
+    TerrainType,
+)
+from ..readers.scenario import (
+    Landscape,
+    Scenario,
+)
+
+
+def classify_scenario(scenario: Scenario) -> Classification:
+    classification = {}
+
+    surface = scenario.map_size[0] * scenario.map_size[1]
+    if surface < 256 * 256:
+        classification["resolution"] = Resolution.LOW
+    elif surface < 1024 * 1024:
+        classification["resolution"] = Resolution.NORMAL
+    else:
+        classification["resolution"] = Resolution.HIGH
+
+    aspect_ratio = scenario.map_size[0] / scenario.map_size[1]
+    if aspect_ratio < 1:
+        aspect_ratio = 1 / aspect_ratio
+
+    if aspect_ratio < 1.2:
+        classification["shape"] = Shape.SQUARE
+    elif aspect_ratio < 2.5:
+        classification["shape"] = Shape.RECTANGLE
+    else:
+        classification["shape"] = Shape.NARROW
+
+    # Only keep the lower heights and skip sea-level.
+    small_histogram = [0] * 16
+    for i in range(1, 15):
+        small_histogram[i] = scenario.histogram[i]
+    for i in range(15, 256):
+        small_histogram[15] += scenario.histogram[i]
+    # Normalize the histogram.
+    small_histogram = [value * 100 / (surface - scenario.histogram[0]) for value in small_histogram]
+
+    # Find all elevations with a certain surface amount. One hill doesn't
+    # make a mountain (yes, this is meant ironically).
+    common_elevations = [i for i, v in enumerate(small_histogram) if v >= 1]
+    # And check the difference between the highest and lowest elevation.
+    height_difference = max(common_elevations) - min(common_elevations)
+
+    if height_difference > 9:
+        classification["terrain-type"] = TerrainType.MOUNTAINOUS
+    elif height_difference > 4:
+        classification["terrain-type"] = TerrainType.HILLY
+    elif height_difference > 1:
+        classification["terrain-type"] = TerrainType.FLAT
+    else:
+        classification["terrain-type"] = TerrainType.VERY_FLAT
```

Comment:
All this code above is identical with heightmap; we should make that common code, so they all identify these things equal.

Reply:
Decided against this, as it is just different enough that it is not really all that useful.